kirin_yacc.py

```diff
@@ -3,4 +3,374 @@
+	'''func_param		: expression more_fpar'''
+
+def p_more_fpar(p):
+	'''more_fpar		: ',' func_param
+									| empty'''
+
+#BLOCK
+def p_block(p):
+	'''block	: '{' bstmt '}' '''
+
+def p_bstmt(p):
+	'''bstmt	: statement bstmt
+						| empty'''
+
+#STATEMENT
+def p_statement(p):
+	'''statement	: assignment
+								| condition
+								| loop
+								| in_out
+								| return
+								| var_decl
+								| CONSTANT var_decl '''
+
+#CONDITION
+def p_condition(p):
+	'''condition	: IF cond_body'''
+
+def p_cond_body(p):
+	'''cond_body	: '(' expression ')' block cond_else'''
+
+def p_cond_else(p):
+	'''cond_else	: ELSE block
+								| ELSEIF cond_body
+								| empty'''
+
+#LOOP
+def p_loop(p):
+	'''loop	: for_loop
+					| while_loop'''
+
+#FOR_LOOP
+def p_for_loop(p):
+	'''for_loop	: FOR '(' assignment expression ';' ID '=' expression ')' block'''
+
+#WHILE_LOOP
+def p_while_loop(p):
+	'''while_loop	: WHILE '(' expression ')' block'''
+
+#IN_OUT
+def p_in_out(p):
+	'''in_out			: PRINT '(' print_exp ')' ';'
+								| SCAN '(' ID id_access ')' ';' '''
+
+def p_print_exp(p):
+	'''print_exp	: print_val print_more'''
+
+def p_print_val(p):
+	'''print_val	: expression
+								| CTE_STRING'''
+
+def p_print_more(p):
+	'''print_more	: ',' print_exp
+								| empty'''
+
+#RETURN
+def p_return(p):
+	'''return		: RETURN ret_val ';' '''
+
+def p_ret_val(p):
+	'''ret_val	: expression
+							| empty'''
+
+#EXPRESSION
+def p_expression(p):
+	'''expression	: rel_expression exp_op'''
+
+def exp_op(p):
+	'''exp_op			: AND expression
+								| OR expression
+								| XOR expression
+								| empty'''
+
+#REL_EXPRESSION_1
+def p_rel_expression(p):
+	'''rel_expression	: rel_expression_1 rel_exp_op'''
+
+def p_rel_exp_op(p):
+	'''rel_exp_op			: EQUAL rel_expression
+										| NOT_EQUAL rel_expression
+										| empty'''
+
+#REL_EXPRESSION_1
+def p_rel_expression_1(p):
+	'''rel_expression_1	: exp rel_exp_1_op'''
+
+def p_rel_exp_1_op(p):
+	'''rel_exp_1_op			: '<' rel_expression_1
+											| LESS_EQUAL_THAN rel_expression_1
+											| '>' rel_expression_1
+											| GREATER_EQUAL_THAN rel_expression_1
+											| empty'''
+
+#EXP
+def p_exp(p):
+	'''exp		: term exp_op'''
+
+def exp_op(p):
+	'''exp_op	: '+' exp
+						| '-' exo
+						| empty'''
+
+#TERM
+def p_term(p):
+	'''term		: factor term_op'''
+
+def p_exp_op(p):
+	'''term_op	: '*' term
+							| '/' term
+							| '%' term
+							| empty'''
+
+#TYPE
+def p_type(p):
+	'''type	: INT
+					| DOUBLE
+					| CHAR
+					| BOOL'''
+
+#VAR_CTE
+def p_var_cte(p):
+	'''var_cte	: CTE_I
+							| CTE_F
+							| CTE_CHAR
+							| CTE_STRING
+							| CTE_BOOL'''
+
+#FACTOR
+def p_factor(p):
+	'''factor		: fact_neg fact_body'''
+
+def p_fact_neg(p):
+	'''fact_neg	: '-'
+							| '~'
+							| empty'''
+
+def p_fact_body(p):
+	'''fact_body	: '(' expression ')'
+								| var_cte
+								| ID fact_id'''
+
+def p_fact_id(p):
+	'''fact_id		: func_call
+								| id_access
+								| empty'''
```

Comment:
Maybe you can add as well the **p_error** rule for displaying an error in the parsing process.
You should also include at the end the yacc.yacc() statement to generate the syntax parser.